src/index.js

```diff
@@ -16,35 +16,26 @@ import * as UrlService from 'hyperview/src/services/url';
 import {
```

Comment:
A general note on file naming / structure:

`index.js` should be reserved for "barrel" modules. "Barrel" modules are files, similar to `__init__.py` in Python, that are used specifically for defining the public API of a particular "sub-module" or section of the package. The barrel module can be used in a couple different ways depending on what the intention is.

For smaller / less complex packages, it can be used just as a way to break up the definition of the public API:

Given the following folder structure:
```
[root]
src
  dom
    index.js
    parser.js
  other-section-a
    ...
    index.js
  other-section-b
    ...
    index.js
  index.js
```

```javascript
// ./src/dom/parser.js
const parser = {...}
export class Parser { ... }

// ./src/dom/index.js
export * from './parser'

// ./src/index.js - root entry point for the package
export * from './dom'
export * from './other-section-a'
export * from './other-section-b'
```
In the above example, the entire public API is available from a single entry point - so the consumer would only ever `import {...} from 'hyperview'`. 

Alternatively, a slightly different pattern can be used to define sub-"namespaces" within a package by effectively defining multiple endpoints: 

```
[root]
dom
  src
    parser.js
  index.js
src
  other-section-b
    ...
    index.js
  other-section-a
    ...
    index.js
  index.js
```

```javascript
// ./dom/src/parser.js
const parser = {...}
export class Parser { ... }

// ./dom/index.js
export * from './src/parser'

// ./src/index.js - root entry point for the package
export * from './other-section-b'
export * from './other-section-b'
```

In this example, the root entry point does _not_ export anything from `dom` itself, instead leaving `dom` to create its own endpoint. In this case, the consumer would `import { Parser } from 'hyperview/dom'` and, `import { everything, else } from 'hyperview'`

Reply:
I agree with most of what's mentioned there. Additionally, we use `import * ` style when possible, which allows for better namespacing of the module public API in the module consumers. i.e. 
```js

import * as Hyperview;

...

Hyperview.Events.dispatch('foo');

```

Reply:
Unless I'm misunderstanding how/where this is getting used, `import * ` is bad form for a browser JS package. It effectively prohibits any tree-shaking of the source, because it forces ALL modules to be loaded and included in the consumer's application bundle.

Bundlers like Webpack (combined with tools like Terser or Uglify) have the ability to remove unused code by inspecting the dependency tree and removing modules that are never accessed. This includes removing unused exports from a single file with multiple exports. However, this only works so long as Webpack can determine that a module is never used - and if modules are imported using `*`, there's nothing to tell Webpack what is actually getting used, so nothing can be "shaken".

more info here: https://webpack.js.org/guides/tree-shaking/

Reply:
Fair warnings and good to learn about this. It appears that metro (the React Native bundler) [does not support tree shaking nor would support it anytime soon](https://github.com/facebook/metro/issues/227#issuecomment-583358386)

Reply:
Oh. Well that's a bummer :(